app/controllers/flavor_controller.rb

```diff
@@ -13,12 +13,55 @@ def self.display_methods
     %w(instances)
   end
 
+  def new
+    assert_privileges('flavor_create')
+    drop_breadcrumb(:name => _("Add a new Flavor"), :url => "/flavor/new")
+    @in_a_form = true
+    @id = 'new'
+  end
+
+  def button
+    if params[:pressed] == 'flavor_create'
+      javascript_redirect(:action => 'new')
+    else
+      delete_flavors
+    end
+  end
+
+  def ems_list
+    assert_privileges('flavor_create')
+    ems_list = ManageIQ::Providers::CloudManager.select do |ems|
+      ems.class::Flavor.supports?(:create)
+    end
+    ems_list.each do |ems|
+      {:name => ems.name, :id => ems.id}
+    end
+    render :json => {:ems_list => ems_list}
+  end
+
   private
 
   def textual_group_list
     [%i(properties relationships), %i(tags)]
   end
   helper_method :textual_group_list
 
+  def delete_flavors
+    assert_privileges('flavor_delete')
+    flavors = find_records_with_rbac(Flavor, checked_or_params)
+    flavors.each do |flavor|
+      begin
+        flavor.delete_flavor_queue(User.current_user.id)
+        add_flash(_("Delete of Flavor \"%{name}\" was successfully initiated.") % {:name => flavor.name})
+      rescue => error
+        add_flash(_("Unable to delete Flavor \"%{name}\": %{details}") % {:name    => flavor.name,
+                                                                          :details => error},
```

Comment:
`error` is an exception, not a message. What do you expect to get displayed here?